### DISCUSSION TITLE
How to config cors with external api ?

### DISCUSSION THREAD

**Discussion by sikeat7** [nuxt/nuxt#16188] [General]
I'm using Nuxt 3 useFetch and I got an error blocked from origin
```
const { data } = await useFetch(fullExternalUrlEndpoint, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: formData
})
```

When passing **mode: 'no-cors'** in config it's return:
Fetch failed loading: POST "https://xxxxxxxxxxxx".

Does anyone know how to config cors correctly ? Thank you in advance

**Comment by Kamsou** [comment]
Hello @sikeat7 

Did you solve your problem? I have the same problem

**Reply by d1i1m1o1n** [reply]
Hello!
In `nuxt.config.ts`

`
vite: {
        server: {
            proxy: {
              '/api/v1': {
                 target: 'https://some.address',
                 changeOrigin: true
              }
        }
    }
`

**Reply by BenjaminOddou** [reply]
hello @d1i1m1o1n, is the `/api/v1` important here ? or can is be something else ? For the target, should it be the full URL of the request ? I still have the CORS issue

```ts
// some page.vue
const config = useRuntimeConfig()
const { data: images } = await useFetch<ImageKit[]>(
  'https://api.imagekit.io/v1/files',
  {
    method: 'GET',
    headers: {
      Accept: 'application/json',
      Authorization: 'Basic ' + config.IMAGEKIT_B64_API
    }
  }
)
```

```ts
// nuxt.config.ts
import { defineNuxtConfig } from 'nuxt'

// https://v3.nuxtjs.org/api/configuration/nuxt.config
export default defineNuxtConfig({
vite: {
    server: {
      proxy: {
        '/v1': {
          target: 'https://api.imagekit.io/v1/files',
          changeOrigin: true
        }
      }
    }
  }
})

**Reply by d1i1m1o1n** [reply]
Try this

```ts
// nuxt.config.ts
import { defineNuxtConfig } from 'nuxt'

// https://v3.nuxtjs.org/api/configuration/nuxt.config
export default defineNuxtConfig({
vite: {
    server: {
      proxy: {
        '/v1/files': {
          target: 'https://api.imagekit.io',
          changeOrigin: true
        }
      }
    }
  }
})
```

**Reply by BenjaminOddou** [reply]
same results unfortunately...

**Reply by d1i1m1o1n** [reply]
I can only share the [documentation](https://vitejs.dev/config/server-options.html#server-proxy), maybe there will be useful information

**Reply by Suniron** [reply]
It works for me, thank's !!! (Nuxt 3 documentation about it is currently down)

**Reply by BenjaminOddou** [reply]
For my case I created a proxy under ˋserver/api/endpoint.ts` and it is working great 🙂

**Reply by DanielTrontsios** [reply]
> For my case I created a proxy under ˋserver/api/endpoint.ts` and it is working great 🙂

I am facing the same issue with cors. Could you please provide the code that fixed your problem? I am not sure what you mean by creating a proxy under server/api

**Reply by BenjaminOddou** [reply]
@JokTronki You can check this [thread](https://github.com/nuxt/framework/issues/7366).

**Comment by Arturexe** [comment]
Having a similar problem trying to send an auth request to laravel backend. Still looking for a solution.
